src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -4,20 +4,21 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
 using Microsoft.PowerFx.Core.Utils;
 
 namespace Microsoft.PowerFx.Core.Public.Types
 {
     public abstract class AggregateType : FormulaType
     {
-        internal AggregateType(DType type) : base(type)
+        internal AggregateType(DType type, Dictionary<string, string> displayNameSet = null) : base(type, displayNameSet)
```

Comment:
>Dictionary [](http://example.com/codeflow?start=43&length=10)

nit: IReadOnlyDictionary<> ?

Reply:
Yes this should be updated -- I don't think we mutate the dictionary, correct?

Reply:
Yup. This should definitely be ReadOnly, good points

Reply:
An abstraction for `BidirectionalDictionary` could be good as well -- which avoids building and managing a hash set in the `FormulaType` constructor. You could use `ImmutableDictionary` which is a persistent form of dictionary, enabling things like adding a property including display name.